src/bin/ion/commands/beta/generate/templates/java/class.templ

```diff
@@ -16,4 +21,110 @@ public final class {{ target_kind_name }} {
+                        {% endif %}
+                     {% else %}
+                        {{ fields[0].value }}.readFrom(reader);
+                     {% endif %}
+        {% elif abstract_data_type is object and abstract_data_type is containing("Structure") %}
+            {# Reads `Structure` class with multiple fields based on `field.name` #}
+            reader.stepIn();
+            while (reader.hasNext()) {
+                reader.next();
+                String fieldName = reader.getFieldName();
+                switch(fieldName) {
+                    {% for field in fields %}
+                    case "{{ field.name }}":
+                        {{ field.name | camel }} = {% if field.value | is_built_in_type %}
+                                                       {% if field.value == "bytes[]" %}
+                                                               reader.newBytes();
+                                                       {% else %}
+                                                            reader.{{ field.value | camel }}Value();
+                                                       {% endif %}
+                                                    {% else %}
+                                                       {{ field.value }}.readFrom(reader);
+                                                    {% endif %}
+                        break;
+                    {% endfor %}
+                    default:
+                        throw new IonException("Can not read field name:" + fieldName + " for {{ target_kind_name }} as it doesn't exist in the given schema type definition.");
+                }
+            }
+            reader.stepOut();
+        {% elif abstract_data_type is object and abstract_data_type is containing("Sequence") %}
+            {# Reads `Sequence` class with a single field `value` that is an `ArraList` #}
+            reader.stepIn();
+            value = new {{ fields[0].value }}();
+            {# Iterate through the `ArraList` and read each element in it based on the data type provided in `abstract_data_type[Sequence]` #}
+            while (reader.hasNext()) {
+                reader.next();
+                {% if abstract_data_type["Sequence"] | is_built_in_type == false %}
+                    value.add({{ abstract_data_type["Sequence"] }}.readFrom(reader));
+                {% else %}
+                    {% if abstract_data_type["Sequence"] == "bytes[]" %}
+                        value.add(reader.newBytes());
+                    {% else %}
+                        value.add(reader.{{ abstract_data_type["Sequence"] | camel }}Value());
+                    {% endif %}
+                {% endif %}
+            }
+            reader.stepOut();
+        {% endif %}
+        return new {{ target_kind_name }}({% for field in fields | sort(attribute="name") -%}{{ field.name | camel }}{% if not loop.last %},{% endif %}{% endfor %});
```

Comment:
We _really_ need to have a builder class so that we can avoid things like sorting the field names and having to check to see if we're on the last loop iteration to see if we need a comma.